parts/k8s/addons/azure-policy-deployment.yaml

```diff
@@ -77,261 +83,243 @@ spec:
-  - get
-  - list
-  - watch
-  - create
   - update
-  - patch
-  - delete
+  - watch
 ---
 apiVersion: rbac.authorization.k8s.io/v1
-kind: ClusterRoleBinding
+kind: RoleBinding
 metadata:
-  creationTimestamp: null
+  labels:
+    gatekeeper.sh/system: "yes"
+    kubernetes.io/cluster-service: "true"
+    addonmanager.kubernetes.io/mode: Reconcile
   name: gatekeeper-manager-rolebinding
+  namespace: kube-system
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: Role
+  name: gatekeeper-manager-role
+subjects:
+- kind: ServiceAccount
+  name: gatekeeper-admin
+  namespace: kube-system
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRoleBinding
+metadata:
   labels:
+    gatekeeper.sh/system: "yes"
     kubernetes.io/cluster-service: "true"
     addonmanager.kubernetes.io/mode: Reconcile
+  name: gatekeeper-manager-rolebinding
 roleRef:
   apiGroup: rbac.authorization.k8s.io
   kind: ClusterRole
   name: gatekeeper-manager-role
 subjects:
 - kind: ServiceAccount
-  name: default
+  name: gatekeeper-admin
   namespace: kube-system
 ---
 apiVersion: v1
 kind: Secret
 metadata:
-  name: gatekeeper-webhook-server-secret
-  namespace: kube-system
   labels:
+    gatekeeper.sh/system: "yes"
     kubernetes.io/cluster-service: "true"
-    addonmanager.kubernetes.io/mode: Reconcile
+    addonmanager.kubernetes.io/mode: EnsureExists
+  name: gatekeeper-webhook-server-cert
+  namespace: kube-system
 ---
 apiVersion: v1
 kind: Service
 metadata:
   labels:
-    control-plane: controller-manager
-    controller-tools.k8s.io: "1.0"
+    gatekeeper.sh/system: "yes"
     kubernetes.io/cluster-service: "true"
     addonmanager.kubernetes.io/mode: Reconcile
-  name: gatekeeper-controller-manager-service
+  name: gatekeeper-webhook-service
   namespace: kube-system
 spec:
   ports:
   - port: 443
     targetPort: 8443
   selector:
     control-plane: controller-manager
-    controller-tools.k8s.io: "1.0"
+    gatekeeper.sh/system: "yes"
 ---
 apiVersion: apps/v1
-kind: StatefulSet
+kind: Deployment
 metadata:
   labels:
     control-plane: controller-manager
-    controller-tools.k8s.io: "1.0"
+    gatekeeper.sh/system: "yes"
     kubernetes.io/cluster-service: "true"
     addonmanager.kubernetes.io/mode: Reconcile
   name: gatekeeper-controller-manager
   namespace: kube-system
 spec:
+  replicas: 1
   selector:
     matchLabels:
       control-plane: controller-manager
-      controller-tools.k8s.io: "1.0"
-  serviceName: gatekeeper-controller-manager-service
+      gatekeeper.sh/system: "yes"
   template:
     metadata:
       labels:
         control-plane: controller-manager
-        controller-tools.k8s.io: "1.0"
+        gatekeeper.sh/system: "yes"
     spec:
       containers:
       - args:
-        - --auditInterval={{ContainerConfig "auditInterval"}}
-        - --constraintViolationsLimit={{ContainerConfig "constraintViolationsLimit"}}
         - --port=8443
+        - --logtostderr
+        - --exempt-namespace=kube-system
```

Comment:
maybe add another issue to track `disable-cert-rotation` and cert generation to work with the addonmanager

Reply:
do we want this to be a part of `rotate-certs`?
https://github.com/Azure/aks-engine/blob/master/docs/topics/certificaterotation.md

Reply:
This sounds separate from rotating cluster certificates, which is what the above doc is meant to describe.

Reply:
we were thinking (as a follow-up) that aks-engine can generate gatekeeper cert and then users can manually rotate using `rotate-certs`. This is similar to behavior in AKS: https://docs.microsoft.com/en-us/azure/aks/certificate-rotation

Reply:
added #2927